Yosemite/Yosemite/Stores/OrderStore.swift

```diff
@@ -384,6 +385,34 @@ private extension OrderStore {
         }
     }
 
+    /// Updates the specified fields from an order optimistically.
+    ///
+    /// Updates will be reverted in case of failure.
+    ///
+    func updateOrderOptimistically(siteID: Int64, order: Order, fields: [OrderUpdateField], onCompletion: @escaping (Result<Order, Error>) -> Void) {
+        // Optimistically update the stored order.
+        let backupOrder = upsertStoredOrder(readOnlyOrder: order)
+
+        remote.updateOrder(from: siteID, order: order, fields: fields) { [weak self] result in
+            guard case .failure = result else {
+                onCompletion(.success(order))
+                return
+            }
+
+            /// Revert optimistic update.
+            ///
+            /// If the backup order is equal to the given order means that the order
+            /// didn't exist locally. So, we have to delete the stored order as workaround.
+            /// Otherwise, we have to revert the updated fields.
+            if order == backupOrder {
```

Comment:
Finally, I've not created a new helper method to extract this logic. I think it doesn't add value and makes the code more complex. WDYT? 🤔

Reply:
This looks good to me 👍